crates/trie/sparse/src/trie.rs

```diff
@@ -129,34 +193,66 @@ impl<P> SparseTrie<P> {
         Ok(self.as_revealed_mut().unwrap())
     }
 
-    /// Wipe the trie, removing all values and nodes, and replacing the root with an empty node.
+    /// Wipes the trie by removing all nodes and values,
+    /// and resetting the trie to only contain an empty root node.
+    ///
+    /// This is useful for when you want to reuse a trie instance without
+    /// creating a new one, clearing all of its contents.
+    ///
+    /// Note: This method will error if the trie is blinded.
     pub fn wipe(&mut self) -> SparseTrieResult<()> {
         let revealed = self.as_revealed_mut().ok_or(SparseTrieErrorKind::Blind)?;
         revealed.wipe();
         Ok(())
     }
 
-    /// Calculates and returns the trie root if the trie has been revealed.
+    /// Calculates the root hash of the trie.
+    ///
+    /// This will update any remaining dirty nodes before computing the root hash.
```

Comment:
can we clarify what does "dirty" mean here? I believe it means that we will also calculate the hashes of all children down to leaves.